File: hw/src/main/java/com/sumian/app/improve/widget/ReportSleepDurationView.java
package com.sumian.app.improve.widget;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.widget.TextView;

import com.sumian.app.R;
import com.sumian.app.common.util.TimeUtil;
import com.sumian.app.improve.widget.base.BaseBlueCardView;

import butterknife.BindView;

/**
 * Created by sm
 * on 2018/3/8.
 * desc:
 */

public class ReportSleepDurationView extends BaseBlueCardView {

    @BindView(R.id.tv_sleep_today_duration)
    TextView mTvSleepTodayDuration;
    @BindView(R.id.light_sleep_duration_view)
    SleepStateDurationView mLightSleepDurationView;
    @BindView(R.id.deep_sleep_duration_view)
    SleepStateDurationView mDeepSleepDurationView;

    public ReportSleepDurationView(@NonNull Context context) {
        this(context, null);
    }

    public ReportSleepDurationView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public ReportSleepDurationView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected int getLayoutRes() {
        return R.layout.hw_lay_sleep_duration_view;
    }

    public void setSleepTodayDuration(int sleepTodayDuration) {
        CharSequence charSequence = TimeUtil.formatSleepDurationText(getContext(), sleepTodayDuration);
        mTvSleepTodayDuration.setText(charSequence);
    }

    public void setDeepSleepData(int deepSleepDuration, int deepSleepDurationPercent) {
        mDeepSleepDurationView.setData("深睡时长", deepSleepDuration, deepSleepDurationPercent);
    }

    public void setLightSleepData(int lightSleepDuration, int lightSleepDurationPercent) {
        mLightSleepDurationView.setData("浅睡时长", lightSleepDuration, lightSleepDurationPercent);
    }

}

File: SumianDeviceDemo/src/main/java/com/sumian/devicedemo/sleepdata/widget/ReportSleepDurationView.kt
package com.sumian.devicedemo.sleepdata.widget

import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.constraintlayout.widget.ConstraintLayout
import com.sumian.devicedemo.R
import kotlinx.android.synthetic.main.hw_lay_sleep_duration_view.view.*

/**
 * Created by sm
 * on 2018/3/8.
 * desc:
 */

class ReportSleepDurationView : ConstraintLayout {

    constructor(context: Context) : this(context, null)
    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
            context,
            attrs,
            defStyleAttr
    ) {
        initView(context)
    }

    private fun initView(context: Context) {
        View.inflate(context, R.layout.hw_lay_sleep_duration_view, this)
    }

    fun setSleepTodayDuration(sleepTodayDuration: Int) {
        tv_sleep_today_duration.setDuration(sleepTodayDuration)
    }

    fun setDeepSleepData(deepSleepDuration: Int, deepSleepDurationPercent: Int) {
        deep_sleep_duration_view.setData(
                resources.getString(R.string.deep_sleep_duration_hint),
                deepSleepDuration,
                deepSleepDurationPercent
        )
    }

    fun setLightSleepData(lightSleepDuration: Int, lightSleepDurationPercent: Int) {
        light_sleep_duration_view.setData(
                resources.getString(R.string.light_sleep_duration_hint),
                lightSleepDuration,
                lightSleepDurationPercent
        )
    }

}
